opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -424,6 +424,8 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
             # Freeze mutable sequences defensively
             if isinstance(value, MutableSequence):
                 value = tuple(value)
+            if isinstance(value, bytes):
+                value = value.decode()
```

Comment:
this could raise an exception due to mis-assuming the string-serializable encoding of the object.

It'd be great if we could choose an encoding that works universally, or maybe just ignore or skip issues: https://docs.python.org/3/library/codecs.html#codecs.decode